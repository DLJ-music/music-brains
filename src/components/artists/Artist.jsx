import React from 'react';
import PropTypes from 'prop-types';

const Artist = ({ name, id, type, score }) => {
  return (
    <>
      <h1 id={id}>{name}</h1>
      {type && <h2>{type}</h2>}
      {score && <h2>{score}</h2>}
    </>
  );
};

Artist.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  type: PropTypes.string,
  score: PropTypes.number
};

export default Artist;

