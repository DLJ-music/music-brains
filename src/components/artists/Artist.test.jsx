import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';

describe('Artist component', () => {
  let artistInfo;
  beforeEach(() => {
    artistInfo = { id: 'stjkstkjlljst', name: 'Jackson Michael', type: 'Group', score: 87 }
  })

  it('renders Artist', () => {
    const wrapper = shallow(<Artist {...artistInfo}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  