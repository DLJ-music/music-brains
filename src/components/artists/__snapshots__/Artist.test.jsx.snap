// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders Artist 1`] = `
<Fragment>
  <h1
    id="stjkstkjlljst"
  >
    Jackson Michael
  </h1>
  <h2>
    Group
  </h2>
  <h2>
    87
  </h2>
</Fragment>
`;
